### ISSUE TITLE
LC0086 False positive

### ISSUE THREAD

**Issue by nickgoddard777** [StefanMaron/BusinessCentral.LinterCop#979]
The new LC0086 is producing a false positive if one of the tests is used somewhere than for page styling. I have a test that is checking if a value from a json file contains the word standard. This is producing the info.

`this.LibraryAssert.AreEqual('Standard', Order."Ship Service Level Category", 'Ship Service Level Category must be filled out');`

**Comment by rvanbekkum** [comment]
Maybe you can consider suppressing the diagnostic for this particular case? I think it's quite rare to use the string-literal `'Standard'`, rather than a translatable text. For the few occurrences that you have, suppressing those cases could be acceptable.

It has been considered/discussed: https://github.com/StefanMaron/BusinessCentral.LinterCop/discussions/805
What are your thoughts on this?

**Comment by nickgoddard777** [comment]
I guess it is rare, but shouldn't the test be more targeted (I have no idea how hard any of this is and am extremely grateful for this brilliant tool).
Do people put page display logic in codeunits, shouldn't they just be in Pages and Page Extensions, and maybe only on statements where variables/labels/Style/StyleExpr are being set by the string rather than just a blanket. Isn't this what you had suggested in the discussion?

**Comment by rvanbekkum** [comment]
> Do people put page display logic in codeunits, shouldn't they just be in Pages and Page Extensions

We often do put the page styling logic in codeunits. Then multiple pages can reuse the same logic.
With that, I think it makes it more difficult to determine whether in the end the string literals are intended to specify a page style or a value for another purpose.

**Comment by Arthurvdv** [comment]
As @rvanbekkum already mentioned, I'm afraid we don't have a way to distinct when the string 'Standard' is used for a page styling of something else.

You can easily suppress this false positive with a pragma.
```AL
#pragma warning disable LC0086 // The value 'Standard' is not used for page styling here
this.LibraryAssert.AreEqual('Standard', Order."Ship Service Level Category", 'Ship Service Level Category must be filled out');
#pragma warning restore LC0086
```

If you don't want to use a pragma you could use a non-Locked label, but there's a possible cavecat here. If there's an translation added for this label, it could then cause unexpected behavior.

```AL
procedure MyProcedure()
var
    StandardLbl: Label 'Standard'; // Do not translate this label ...
begin
    this.LibraryAssert.AreEqual(StandardLbl, Order."Ship Service Level Category", 'Ship Service Level Category must be filled out');
end;
```

Locking the Label will raise the LC0086 again, so you then have to add a pragma again (and we're kinda back where we started).

```AL
procedure MyProcedure()
var
#pragma warning disable LC0086 // The value 'Standard' is not used for page styling
    StandardLbl: Label 'Standard', Locked = true;
#pragma warning restore LC0086
begin
    this.LibraryAssert.AreEqual(StandardLbl, Order."Ship Service Level Category", 'Ship Service Level Category must be filled out');
end;
```

The benefit of this could be that you can maybe have this as a global variable, and use this in different methodes. Then you only have to set the pragma once on the label.
